code/game/objects/structures/training_machine.dm

```diff
@@ -0,0 +1,310 @@
+	if (!attached_item)
+		return
+	vis_contents.Cut()
+	if (istype(attached_item, /obj/item/storage/toolbox/syndicate))
+		qdel(attached_item)
+	else if (user)
+		user.put_in_hands(attached_item)
+	else
+		attached_item.forceMove(drop_location())
+	if (throwing && !QDELETED(attached_item)) //Fun little thing where we throw out the old attached item when emagged
+		//We do a QDELETED check here because we don't want to throw the syndi toolbox, if it exists
+		var/destination = get_edge_target_turf(get_turf(src), pick(GLOB.alldirs))
+		attached_item.throw_at(destination, 4, 1)
+	attached_item = null
+	handle_density()
+
+/obj/structure/training_machine/AltClick(mob/user)
+	. = ..()
+	if (!attached_item)
+		return
+	if (obj_flags & EMAGGED)
+		to_chat(user, "<span class='warning'>The toolbox is somehow stuck on! It won't budge!</span>")
+		return
+	to_chat(user, "<span class='notice'>You remove \the [attached_item] from the training device.</span>")
+	remove_attached_item(user)
+	playsound(src, "rustle", 50, TRUE)
+
+/obj/structure/training_machine/proc/toggle()
+	if (moving)
+		moving = FALSE
+	else
+		start_moving()
+
+/obj/structure/training_machine/proc/stop_moving(message = "Ending training simulation.")
+	moving = FALSE
+	starting_turf = null
+	say(message)
+	playsound(src,'sound/machines/synth_no.ogg',50,FALSE)
+
+/obj/structure/training_machine/proc/start_moving()
+	moving = TRUE
+	starting_turf = get_turf(src)
+	say("Beginning training simulation.")
+	playsound(src,'sound/machines/triple_beep.ogg',50,FALSE)
+	do_movement()
+
+/obj/structure/training_machine/proc/do_movement()
+	var/turf/current_turf = get_turf(src)
+	if (!moving || !starting_turf || isspaceturf(current_turf))
+		stop_moving()
+		return
+	if (current_turf == target_position) //We've reached our target turf, now find a new one
+		target_position = null
+	if (!target_position)
+		target_position = find_target_position()
+		if (!target_position)
+			stop_moving("ERROR! Cannot calculate suitable movement path.")
+	var/turf/nextStep = get_step_towards(src, target_position)
+	if (!Move(nextStep, get_dir(src, nextStep)))
+		target_position = null //We couldn't move towards the target turf, so find a new target turf
+	try_attack()
+	addtimer(CALLBACK(src, .proc/do_movement), max(MAX_SPEED - move_speed, 1)) // We want to ensure this is never <=0
+
+/obj/structure/training_machine/proc/find_target_position()
+	var/list/turfs = list()
+	for(var/turf/potential_turf in view(range, starting_turf))
+		if (potential_turf.is_blocked_turf() || potential_turf == target_position)
+			continue
+		turfs += potential_turf
+	if (!length(turfs))
+		return
+	return pick(turfs)
+
+/obj/structure/training_machine/proc/try_attack()
+	if (!attached_item || istype(attached_item, /obj/item/target))
+		return
+	if (world.time < last_attack_time + attack_cooldown)
+		return
+	var/list/targets
+	for(var/mob/living/carbon/target in oview(1, get_turf(src))) //Find adjacent target
```

Comment:
```suggestion
	for(var/mob/living/carbon/target in view(1, get_turf(src))) //Find adjacent target
```
Do you want to ignore targets which might happen to be on the same tile (teleported, getting there before the mahine became dense, or who knows how else)?

Reply:
Yes, otherwise it will hit mobs buckled to it. I want to train against a monkey buckled to a target stake, after all.
Also, what would be the best way to get a list of mobs adjacent to this, if not oview()?

Reply:
Ah, then it's fine.